### PR TITLE
Compile & Helper: Use cmake for project with cmake available.

### PR DESCRIPTION
**Pull request by 1480c1**
Trying out this Draft Pull Request thing.
Trying out using cmake when possible.
Tested libtiff and libwebp and works so far.
Going through the rest of the packages and seeing what has a cmakelist.txt file to see if they can use cmake.

### CODE REVIEW

**Comment by 1480c1** [comment]
Supposedly, this makes it so that you can open a discussion with proposed code without it being interpreted as final code. Probably would have been useful for the no Mintty thing.

**Comment by 1480c1** [comment]
Probably libtiff is unnecessary, although I was having problems building webp, tesseract, and leptonica earlier using the one generated using do_pacman_install libtiff  fix_libtiff_pc. Going to try without libtiff build.

**Comment by 1480c1** [comment]
Doesn't seem to work with pacman libtiff.

**Comment by wiiaboo** [comment]
Problem with those where cmake is optional is that usually they don't have static-only builds.
And in that point, why compile libtiff shared?

**Comment by 1480c1** [comment]
I'm not exactly sure what webp, leptonica, and tesseract needs from libtiff so I'm compiling it shared.

**Comment by 1480c1** [comment]
Do you know which one specifically did not offer static-only builds? I'll try to see if they are still non-static only or ignore those if not possible.

**Comment by 1480c1** [comment]
So, question, why do we use `-DUNIX=on` when there is a `-DMINGW=on`?

**Comment by 1480c1** [comment]
I'm seeming to have problems with tesseract, its complaining about avx right now. Need to figure out why.

**Comment by wiiaboo** [comment]
> 
> 
> So, question, why do we use `-DUNIX=on` when there is a `-DMINGW=on`?

It's needed in a few projects to generate a pkg-config or use pkg-config.

**Comment by 1480c1** [comment]
I see, main reason why I ask is because some packages failed to build due to the DUnix flag, thinking that there were some specific headers available.

**Comment by 1480c1** [comment]
Next question, why do we use `https://notabug.org/RiCON/soxr.git` instead of `https://git.code.sf.net/p/soxr/code`? Is there something different between the two? Running `diff -ur` between cloned repos of the two only show stuff in the .git folder.

**Comment by wiiaboo** [comment]
sf.net git was kind of unstable at the time. notabug is being used as a mirror. Same as with mxf libs.

**Comment by schmidthubert** [comment]
Using cmake for sndfile (even without any params) also fixes https://github.com/jb-alvarado/media-autobuild_suite/issues/1077

I've just tested it under Win10 Home 32bit :+1:

**Comment by 1480c1** [comment]
Then I will create a separate pr with the libsndfile with cmake.

File: build/media-suite_compile.sh [hunk 1]
```diff
@@ -1420,7 +1420,7 @@ EOF
     fi
     }
     build_x265
-    log "install" ninja -j1 install
+    log "install" ninja install
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: Ninja sometimes tries to install faster than the hard drives/file system can handle. File creation/copy should remain single-threaded when possible.

Comment (1480c1) [reply build/media-suite_compile.sh]: Ah, I just realized that was an install, will fix later.

**Comment by 1480c1** [comment]
Right now testing with a already set up suite then going to test with a fresh one, so far it seems ok.

**Comment by 1480c1** [comment]
What happened to hubert? He turned into a ghost

**Comment by wiiaboo** [comment]
Maybe tired or offended by something I/someone said. Not the first time
I've seen it happen in Github.

**Pull request closed without merging**